snapcraft/plugins/dotnet.py

```diff
@@ -140,3 +140,10 @@ def _get_appname(self):
             if fnmatch.fnmatch(file, '*.??proj'):
                 return os.path.splitext(file)[0]
                 break
+
+    def env(self, root):
+        # Update the PATH only during the Build and Install step
```

Comment:
Thanks for adding the comment, but it still leaves me asking: why? Why not during the pull step as well?

Reply:
this should be ok (although overriding pull is not possible).

Reply:
I'm still confused why this special case is needed. It doesn't matter if the dir doesn't exist:

```
$ ls -l /tmp/foo
ls: cannot access '/tmp/foo': No such file or directory
$ export PATH=/tmp/foo:$PATH
$ which footest
$ mkdir /tmp/foo
$ touch /tmp/foo/footest
$ chmod a+x /tmp/foo/footest 
$ which footest
/tmp/foo/footest
```

Reply:
Ah, I am indeed missing something-- this is to avoid having this PATH definition make it into the final snap. We need to support this better, but this is a good workaround for now, so you can ignore me.